faker/providers/color/color.py

```diff
@@ -0,0 +1,276 @@
+"""
+Code adapted from:
+- https://github.com/davidmerfield/randomColor  (CC0)
+- https://github.com/kevinwuhoo/randomcolor-py  (MIT License)
+
+Additional reference from:
+- https://en.wikipedia.org/wiki/HSL_and_HSV
+"""
+
+import colorsys
+import math
+import random
+import sys
+
+import six
+
+
+COLOR_MAP = {
+    'monochrome': {
+        'hue_range': [0, 0],
+        'lower_bounds': [
+            [0, 0], [100, 0],
+        ],
+    },
+    'red': {
+        'hue_range': [-26, 18],
+        'lower_bounds': [
+            [20, 100], [30, 92], [40, 89],
+            [50, 85], [60, 78], [70, 70],
+            [80, 60], [90, 55], [100, 50],
+        ],
+    },
+    'orange': {
+        'hue_range': [19, 46],
+        'lower_bounds': [
+            [20, 100], [30, 93], [40, 88], [50, 86],
+            [60, 85], [70, 70], [100, 70],
+        ],
+    },
+    'yellow': {
+        'hue_range': [47, 62],
+        'lower_bounds': [
+            [25, 100], [40, 94], [50, 89], [60, 86],
+            [70, 84], [80, 82], [90, 80], [100, 75],
+        ],
+    },
+    'green': {
+        'hue_range': [63, 178],
+        'lower_bounds': [
+            [30, 100], [40, 90], [50, 85], [60, 81],
+            [70, 74], [80, 64], [90, 50], [100, 40],
+        ],
+    },
+    'blue': {
+        'hue_range': [179, 257],
+        'lower_bounds': [
+            [20, 100], [30, 86], [40, 80],
+            [50, 74], [60, 60], [70, 52],
+            [80, 44], [90, 39], [100, 35],
+        ],
+    },
+    'purple': {
+        'hue_range': [258, 282],
+        'lower_bounds': [
+            [20, 100], [30, 87], [40, 79],
+            [50, 70], [60, 65], [70, 59],
+            [80, 52], [90, 45], [100, 42],
+        ],
+    },
+    'pink': {
+        'hue_range': [283, 334],
+        'lower_bounds': [
+            [20, 100], [30, 90], [40, 86], [60, 84],
+            [80, 80], [90, 75], [100, 73],
+        ],
+    },
+}
+
+
+class RandomColor(object):
+
+    def __init__(self, generator=None, seed=None):
+        self.colormap = COLOR_MAP
+
+        # Option to specify a seed was not removed so this class
+        # can still be tested independently w/o generators
+        if generator:
+            self.random = generator.random
+        else:
+            self.seed = seed if seed else random.randint(0, sys.maxsize)
+            self.random = random.Random(self.seed)
+
+        for color_name, color_attrs in self.colormap.items():
+            lower_bounds = color_attrs['lower_bounds']
+            s_min = lower_bounds[0][0]
+            s_max = lower_bounds[len(lower_bounds) - 1][0]
```

Comment:
isn't this equivalent to `s_max = lower_bounds[-1][0]`?

Reply:
You are right. I wonder why the author of `randomcolor-py` did not just use `[-1]` for this one. I will fix this, and I am sorry I did not notice.